app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,44 @@
+class MerchantsController < ApplicationController
+
+  def index
+    @merchants = Merchant.all
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def item_index
+    @merchant = Merchant.find(params[:merchant_id])
+  end
```

Comment:
This is a fine thing to do. You may also consider making the merchant/:merchant_id/items route to the items#index action and then assign @ merchant in that method if the param :merchant_id comes through. Then you can reuse the same view since they are pretty much identical except for a link or two.